code/game/objects/machinery/cryopod.dm

```diff
@@ -172,9 +172,34 @@
 
 		visible_message(span_notice("[src] beeps happily as it disgorges the desired objects."))
 
-		for(var/obj/item/I in GLOB.cryoed_item_list_gun || GLOB.cryoed_item_list_ammo || GLOB.cryoed_item_list_explosive || GLOB.cryoed_item_list_melee || GLOB.cryoed_item_list_clothing || GLOB.cryoed_item_list_food || GLOB.cryoed_item_list_drugs || GLOB.cryoed_item_list_containers || GLOB.cryoed_item_list_other)
-			dispense_item(I, usr, FALSE)
-
+		switch(state)
+			if(STATE_GUN)
+				for(var/obj/item/I in GLOB.cryoed_item_list_gun)
+					dispense_item(I, usr, FALSE)
+			if(STATE_AMMO)
+				for(var/obj/item/I in GLOB.cryoed_item_list_ammo)
+					dispense_item(I, usr, FALSE)
+			if(STATE_EXPLOSIVE)
+				for(var/obj/item/I in GLOB.cryoed_item_list_explosive)
+					dispense_item(I, usr, FALSE)
+			if(STATE_MELEE)
+				for(var/obj/item/I in GLOB.cryoed_item_list_melee)
+					dispense_item(I, usr, FALSE)
+			if(STATE_CLOTHING)
+				for(var/obj/item/I in GLOB.cryoed_item_list_clothing)
+					dispense_item(I, usr, FALSE)
+			if(STATE_FOOD)
+				for(var/obj/item/I in GLOB.cryoed_item_list_food)
+					dispense_item(I, usr, FALSE)
+			if(STATE_DRUGS)
+				for(var/obj/item/I in GLOB.cryoed_item_list_drugs)
+					dispense_item(I, usr, FALSE)
+			if(STATE_CONTAINERS)
+				for(var/obj/item/I in GLOB.cryoed_item_list_containers)
+					dispense_item(I, usr, FALSE)
+			if(STATE_OTHER)
+				for(var/obj/item/I in GLOB.cryoed_item_list_other)
+					dispense_item(I, usr, FALSE)
```

Comment:
just switch to change what list to dispense from. No point in copypasting the dispense 10 times.